chart/templates/default-transit-auth-method.yaml

```diff
@@ -18,14 +18,40 @@ metadata:
 spec:
   vaultConnectionRef: {{ .Values.controller.manager.clientCache.storageEncryption.vaultConnectionRef }}
   namespace: {{ .Values.controller.manager.clientCache.storageEncryption.namespace }}
-  method: "kubernetes"
+  method: {{ .Values.controller.manager.clientCache.storageEncryption.method }}
   mount: {{ .Values.controller.manager.clientCache.storageEncryption.mount }}
+  {{- if eq .Values.controller.manager.clientCache.storageEncryption.method "kubernetes" }}
   kubernetes:
-    role: {{ .Values.controller.manager.clientCache.storageEncryption.role }}
-    serviceAccount: {{ .Values.controller.manager.clientCache.storageEncryption.serviceAccount | default (printf "%s-controller-manager" (include "chart.fullname" . )) }}
-    {{- if .Values.controller.manager.clientCache.storageEncryption.tokenAudiences }}
-    audiences: {{ .Values.controller.manager.clientCache.storageEncryption.tokenAudiences | toJson }}
+    role: {{ .Values.controller.manager.clientCache.storageEncryption.kubernetes.role }}
+    serviceAccount: {{ .Values.controller.manager.clientCache.storageEncryption.kubernetes.serviceAccount | default (printf "%s-controller-manager" (include "chart.fullname"
+     . )) }}
+    {{- if .Values.controller.manager.clientCache.storageEncryption.kubernetes.tokenAudiences }}
+    audiences: {{ .Values.controller.manager.clientCache.storageEncryption.kubernetes.tokenAudiences | toJson }}
     {{- end }}
+  {{- else if eq .Values.controller.manager.clientCache.storageEncryption.method "jwt" }}
+  jwt:
```

Comment:
At first glance this configuration seems pretty similar to what we have in `default-vault-auth-methods.yaml`. I wonder if there could be a way to share the rendering tasks between the two files?

Reply:
Good point, I was planning on creating a function for it, but got stuck on the templating logic, fixed in 18a15e7